### PR TITLE
WT-7267 Compare entire history store key when inferring cursor position in `search_near`

### PR DESCRIPTION
**Pull request by tetsuo-cpp**
I have a WiredTiger patch build (incl. stress tests) + about 2k successful runs on my spawn hosts. I'm going to continue running testing to gain confidence, but I think we can begin reviewing and looking at getting this merged.

### CODE REVIEW

File: src/cursor/cur_hs.c [hunk 1]
```diff
@@ -651,9 +651,17 @@ __curhs_search_near(WT_CURSOR *cursor, int *exactp)
...
+                     * store component) because ordering can be different between the data store and
+                     * history store due to packing. Since we know we're NOT in the specified key
+                     * range due to the check above, checking whether we're before or after the full
+                     * history store key that we're running a `search near` on will tell us whether
+                     * we're before or after the specified key range.
+                     *
+                     * If we're before the specified key range, that means nothing is visible to us
+                     * in the specified key range and we should terminate the search.
                      */
+                    WT_ERR(__wt_compare(session, NULL, &file_cursor->key, srch_key, &cmp));
```
Comment (quchenhao) [inline src/cursor/cur_hs.c:664]: The fix makes sense but we have to do a full comparison again worries me. Can we run a mongodb perf test to see if this has any impact?

Comment (tetsuo-cpp) [reply src/cursor/cur_hs.c:664]: Good point. I'll kick some off this afternoon.

**Review by keithbostic — APPROVED** [review]
lgtm

**Comment by tetsuo-cpp** [comment]
@quchenhao WiredTiger/MongoDB patch builds + SysPerf are all looking pretty green. Let's look at getting this in today.

**Review by quchenhao — APPROVED** [review]
lgtm

**Pull request merged**